projects/packages/forms/src/contact-form/js/grunion-admin.js

```diff
@@ -233,16 +233,37 @@ jQuery( function ( $ ) {
 		}, 5000 );
 	}
 
-	$( document ).on( 'click', '#jetpack-form-responses-connect', function () {
+	$( document ).on( 'click', '#jetpack-form-responses-connect', function ( e ) {
 		const $this = $( this );
 		const name = $this.data( 'nonce-name' );
 		const value = $( '#' + name ).attr( 'value' );
+		const redirectUrl = $this.attr( 'href' );
+
+		// Pause redirect for tracking.
+		e.preventDefault();
+
 		$this.attr( 'disabled', 'disabled' );
 		$this.text(
 			( window.exportParameters && window.exportParameters.waitingConnection ) ||
 				'Waiting for connection...'
 		);
+
+		// Do the tracking.
+		if (
+			'undefined' !== typeof analytics &&
+			'undefined' !== typeof jetpack_forms_tracking &&
+			jetpack_forms_tracking?.tracksUserData
+		) {
+			const tracksUser = jetpack_forms_tracking.tracksUserData;
+			analytics.initialize( tracksUser.userid, tracksUser.username );
+			analytics.tracks.recordEvent( 'jetpack_forms_upsell_googledrive_click', {
+				screen: 'form-submissions-export',
+			} );
+		}
+
+		// Start polling and continue redirect.
 		startPollingConnection( { name, value } );
+		window.open( redirectUrl, '_blank' );
```

Comment:
@jeherve I updated the implementation here to use Tracking::register_tracks_functions_scripts(). Above is the associated JS. I found the event was not firing in Tracks Vigilante. It seemed to be because the page quickly redirects to WordPress.com to connect Google. 

If I add e.preventDefault() like above, and manually do the redirect at the end with window.open(), I DO see the event fired. Would you expect this kind of imposed-pause to be needed for tracking?

Reply:
"it didn't work either with your first implementation that added the extra attributes directly to the PHP markup?" -> I didn't get the first implementation to work at all. But I didn't dig deeply. I had already decided after a bit I'd try this approach, which seems newer and well suited to our needs. 

I'll double check this and be sure I'm not missing something, but otherwise I proceed with this approach unless there are objections. 

It's worth noting that we're planning to deprecate and ideally remove this page and just use the other, react-based page for viewing form submissions. So it's not worth a lot of time to troubleshoot this.

Reply:
Follow up: for now, I'm planning to leave this in places as it appears to be needed to work. If anyone has concerns, let me know. Again, this page will likely go away or at least be very deemphasized in the coming months in any case.